test/index.js

```diff
@@ -2989,6 +3003,18 @@ describe('def', function() {
 
     eq(bimap.toString(), 'bimap :: Bifunctor f => (a -> b) -> (c -> d) -> f a c -> f b d');
     eq(bimap(length, Math.sqrt, Pair('Sanctuary', 25)), Pair(9, 5));
+
+    throws(function() { bimap(length, Math.sqrt, Pair(['foo', true, 42], null)); },
+           TypeError,
+           'Type-variable constraint violation\n' +
+           '\n' +
+           'bimap :: Bifunctor f => (a -> b) -> (c -> d) -> f a c -> f b d\n' +
+           '                                                  ^\n' +
+           '                                                  1\n' +
+           '\n' +
+           '1)  ["foo", true, 42] :: Array ???\n' +
+           '\n' +
+           'Since there is no type of which all the above values are members, the type-variable constraint has been violated.\n');
```

Comment:
If I implement `showType` as such:

```js
  //  showType :: Type -> String
  function showType(t) {
    return isConsistent(t) ?
           unless(t.type === FUNCTION || t.type === RECORD || isEmpty(t.keys),
                  stripOutermostParens,
                  String(t)) :
           '(inconsistent)';
  }
```

Then any inconsistent type will simply be rendered as `(inconsistent)`, this removes the confusion between values for which the type has not been determined (such as the inner type of a `Nothing`), and values determined to be inconsistent and therefore members of no types. For me this is clearer, but I also realize now how minor the issue is.

Another approach could be to change `always('???')` to `always('(inconsistent)')` as the render function of Inconsistent, which would make this type render as `Array (inconsistent)`. This also alleviates some confusion, but then we leave the fact that an array of inconsistent types is "not an actual type" implied.